File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-5.py
class Solution:
    def characterReplacement(self, string: str, k: int) -> int:
        count = {}
        result = 0
        left = 0

        for right in range(len(string)):
            count[string[right]] = 1 + count.get(string[right], 0)

            while (right - left) + 1 - max(count.values()) > k:
                count[string[left]] -= 1
                left += 1
             
            result = max(result, (right - left) + 1)
        
        return result

File: Data Structures & Algorithms/max-water-container/submission-5.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0;
        int right = heights.size() - 1;
        int maxArea = 0;

        while (left < right){
            
            maxArea = max((min(heights[left], heights[right])) * (right - left), maxArea);

            if (heights[left] < heights[right]){
                left += 1;
            } else {
                right -= 1;
            }
        }

        return maxArea;
    }
};
